test/systemtests/util_test.go

```diff
@@ -1328,36 +1367,50 @@ func (s *systemtestSuite) SetUpTestBaremetal(c *C) {
 }
 
 func (s *systemtestSuite) SetUpTestVagrant(c *C) {
-	for _, node := range s.nodes {
+	s.parallelExec(func(node *node) error {
 		node.exec.cleanupContainers()
 		node.stopNetplugin()
 		node.cleanupSlave()
-	}
+		return nil
+	})
 
-	for _, node := range s.nodes {
+	s.parallelExec(func(node *node) error {
 		node.stopNetmaster()
-
-	}
-	for _, node := range s.nodes {
 		node.cleanupMaster()
-	}
+		return nil
+	})
 
-	for _, node := range s.nodes {
+	errors := s.parallelExec(func(node *node) error {
+		return node.startNetplugin("")
 
-		c.Assert(node.startNetplugin(""), IsNil)
-		c.Assert(node.exec.runCommandUntilNoNetpluginError(), IsNil)
+	})
+	for _, err := range errors {
+		c.Assert(err, IsNil)
+	}
 
+	errors = s.parallelExec(func(node *node) error {
+		return node.exec.runCommandUntilNoNetpluginError()
+	})
+	for _, err := range errors {
+		c.Assert(err, IsNil)
 	}
 
-	time.Sleep(15 * time.Second)
+	time.Sleep(5 * time.Second)
```

Comment:
I don't think we should remove these delays unless we add a polling to test for the status we are expecting. These delays were added to address failures due to timing differences between machines.